app.py

```
@@ -43,9 +53,25 @@ def POST_makeMosaic():
         input_file.save(base_img_name)
 
         for idx, (theme, mg_url) in enumerate(mg_ports.items(), 1):
-            print(f"Generating {theme} mosiac ({idx}/{len(mg_ports)})")
+            print(f"Generating {theme} mosaic ({idx}/{len(mg_ports)})")
+            tilesAcross = request.form["tilesAcross"]
+            renderedTileSize = request.form["renderedTileSize"]
+            """
+            Checking the input for tilesAcross and renderedTileSize, having conditions below will set the value to default:
+            1) Input is not exist
+            2) Negative Input
+            3) tilesAcross or renderedTileSize is too huge
+            4) Non-integer input
+            """
+            if tilesAcross == '' or tilesAcross == None or float(tilesAcross) <= 0 or float(tilesAcross) > MAX_TILES_ACORSS or float(tilesAcross) != int(float(tilesAcross)):
+                tilesAcross = TILES_ACORSS
+            if renderedTileSize == '' or renderedTileSize == None or float(renderedTileSize) <= 0 or float(renderedTileSize) > MAX_RENDERED_TILE_SIZE or float(renderedTileSize) != int(float(renderedTileSize)):
+                renderedTileSize = RENDERED_TILE_SIZE
+            if tilesAcross * renderedTileSize > MAX_PIXEL:
+                tilesAcross = TILES_ACORSS
+                renderedTileSize = RENDERED_TILE_SIZE
```

We should return an error, instead of assuming a specific value for the user.